domain/application/service/provider.go

```diff
@@ -251,6 +251,20 @@ func (s *ProviderService) SetApplicationConstraints(
 	return s.st.SetApplicationConstraints(ctx, appID, constraints.DecodeConstraints(cons))
 }
 
+// GetApplicationStorageDirectives returns the storage directives that are
+// set for an application. If the application does not have any storage
+// directives set then an empty result is returned.
+//
+// The following error types can be expected:
+// - [github.com/juju/juju/domain/application/errors.ApplicationNotFound]
+// when the application no longer exists.
+func (s *ProviderService) GetApplicationStorageDirectives(
+	ctx context.Context,
+	uuid coreapplication.UUID,
+) ([]application.StorageDirective, error) {
```

Comment:
Whatever this func ends up becoming based on PR feedback the caller needs it as a map. Help them out.